src/scheduler.cc

```diff
@@ -902,9 +910,20 @@ void Scheduler::tick(Locker& locker, int64 now) {
   for (SchedulerThread* thread : threads_) {
     Process* process = thread->interpreter()->process();
     if (process == null) continue;
+    uint64 us_since_preemption = now - process->run_timestamp();
+    if (process->signals() & Process::PREEMPT) {
+      // The process is already suppossed to preempt.
+      // Check whether it is stuck.
+      if (us_since_preemption > MAX_RUN_WITHOUT_PREEMPTION_US) {
+        FATAL("Process is not yielding\n");
```

Comment:
Maybe we can print the process better here? Even %p -- or perhaps better process->id() -- could be useful.

Reply:
done.